wagtail/images/models.py

```diff
@@ -300,93 +309,166 @@ def get_rendition_model(cls):
         """Get the Rendition model for this Image model"""
         return cls.renditions.rel.related_model
 
-    def get_rendition(self, filter):
+    def get_rendition(self, filter: Union["Filter", str]) -> "AbstractRendition":
+        """
+        Returns a ``Rendition``* instance with a ``file`` field value (an
+        image) reflecting the supplied ``filter`` value and focal point values
+        from this object.
+
+        *If using custom image models, an instance of the custom rendition
+        model will be returned.
+        """
         if isinstance(filter, str):
             filter = Filter(spec=filter)
 
-        cache_key = filter.get_cache_key(self)
         Rendition = self.get_rendition_model()
 
         try:
-            rendition_caching = True
+            rendition = self.find_existing_rendition(filter)
+        except Rendition.DoesNotExist:
+            rendition = self.create_rendition(filter)
+            # Reuse this rendition if requested again from this object
+            if "renditions" in getattr(self, "_prefetched_objects_cache", {}):
+                self._prefetched_objects_cache["renditions"]._result_cache.append(
+                    rendition
+                )
+
+        try:
             cache = caches["renditions"]
-            rendition_cache_key = Rendition.construct_cache_key(
-                self.id, cache_key, filter.spec
+            key = Rendition.construct_cache_key(
```

Comment:
non-blocking nitpick: we use this in three different places.
perhaps candidate for an internal method that takes the filter as a param?